File: src/main/java/no/nav/syfo/config/OppgaveClientConfigProvider.java
package no.nav.syfo.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class OppgaveClientConfigProvider {

    @Bean
    public OppgaveConfig getOppgaveClientConfig(
        @Value("${oppgavebehandling.url}")
        String url,
        @Value("${securitytokenservice.url}")
        String tokenUrl,
        @Value("${srvsyfoinntektsmelding.username}")
        String username,
        @Value("${srvsyfoinntektsmelding.password}")
        String password) {
        return new OppgaveConfig(url, tokenUrl, username, password);
    }

}

File: src/main/kotlin/no/nav/syfo/config/OppgaveClientConfigProvider.kt
package no.nav.syfo.config

data class OppgaveConfig(val url: String, val tokenUrl: String, val username: String, val password: String)

class OppgaveClientConfigProvider(private val url: String,
                                  private val tokenUrl: String,
                                  private val username: String,
                                  private val password: String){

    fun getOppgaveClientConfig(): OppgaveConfig {
        return OppgaveConfig(url, tokenUrl, username, password)
    }
}


